zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanContextUtil.java

```diff
@@ -16,19 +16,33 @@
 
 package com.linecorp.armeria.internal.tracing;
 
+import java.util.concurrent.TimeUnit;
+
 import com.linecorp.armeria.common.logging.RequestLog;
 
 import brave.Span;
 
 public final class SpanContextUtil {
 
     /**
-     * Adds logging tags to the provided {@link Span} and closes it.
+     * Starts the {@link Span} when the log is ready.
+     */
+    public static void startSpan(Span span, RequestLog log) {
+        span.start(log.requestStartTimeMicros());
+    }
+
+    /**
+     * Adds logging tags to the provided {@link Span} and closes it when the log is finished.
      * The span cannot be used further after this method has been called.
      */
     public static void closeSpan(Span span, RequestLog log) {
         SpanTags.addTags(span, log);
-        span.finish();
+        span.finish(wallTimeMicros(log, log.responseEndTimeNanos()));
+    }
+
+    private static long wallTimeMicros(RequestLog log, long timeNanos) {
+        long relativeTimeNanos = timeNanos - log.requestStartTimeNanos();
```

Comment:
nit: is this `log.totalDurationNanos()`?

Reply:
Currently yes since I only pass `responseEndTimeNanos()` into this function. Later I want to also pass `requestFirstBytesTransferredNanos()` and `responseFirstBytesTransferredNanos()`. Is it clearer if I use that now and replace it with a generic approach later?

Reply:
Oh I see. Now I get it. Fine as it is.

Reply:
So am I :)